pkg/tests/example_transpile_test.go

```diff
@@ -121,15 +131,18 @@ func TestGenerateExamples(t *testing.T) {
 	// memory consumption of using independent schema.Package instances, we
 	// decided to maintain a single copy of each schema until non-determinism is
 	// observed.
-	var generateLock sync.Mutex
```

Comment:
If your going to dispose of the lock:
1. I want to know what has changed that this won't cause occasional race conditions (and thus flakey tests).
2. We should remove the comment explaining the lock if we remove the lock.

Reply:
The primary issue with the lock was `yarn install`, and I kept it for wrapping the `TypeCheck` function exclusively.

Agree on removing the comment :)

Reply:
I also think my related changes upstream & to use GenerateProject solved the concurrency issues, could not reproduce locally with high `-parallelism`.

Reply:
I remember that it had something to do with binding the same schema in parallel, but your inability to replicate indicates that I misremembered.